packages/features/ee/teams/components/TeamEventTypeForm.tsx

```diff
@@ -0,0 +1,190 @@
+import { zodResolver } from "@hookform/resolvers/zod";
+import type { EventType } from "@prisma/client";
+import type { ReactNode } from "react";
+import { useEffect } from "react";
+import { useForm } from "react-hook-form";
+import type { z } from "zod";
+
+import { useOrgBranding } from "@calcom/features/ee/organizations/context/provider";
+import { classNames } from "@calcom/lib";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { HttpError } from "@calcom/lib/http-error";
+import slugify from "@calcom/lib/slugify";
+import { SchedulingType } from "@calcom/prisma/enums";
+import { unlockedManagedEventTypeProps } from "@calcom/prisma/zod-utils";
+import { createEventTypeInput } from "@calcom/prisma/zod/custom/eventtype";
+import { trpc } from "@calcom/trpc/react";
+import { Form, TextField, showToast, Tooltip } from "@calcom/ui";
+import { Alert, RadioGroup as RadioArea } from "@calcom/ui";
+
+type props = {
+  isAdmin: boolean;
+  teamId: number;
+  handleSuccessMutation: (eventType: EventType) => void;
+  submitButton: (isPending: boolean) => ReactNode;
+};
+export const TeamEventTypeForm = ({ isAdmin, teamId, handleSuccessMutation, submitButton }: props) => {
+  const { t } = useLocale();
+  const utils = trpc.useUtils();
+  const orgBranding = useOrgBranding();
+  const { data: team } = trpc.viewer.teams.get.useQuery({ teamId, isOrg: false }, { enabled: !!teamId });
+  const form = useForm<z.infer<typeof createEventTypeInput>>({
+    defaultValues: {
+      length: 15,
+    },
+    resolver: zodResolver(createEventTypeInput),
+  });
+
+  const schedulingTypeWatch = form.watch("schedulingType");
+  const isManagedEventType = schedulingTypeWatch === SchedulingType.MANAGED;
+  const urlPrefix = orgBranding?.fullDomain ?? process.env.NEXT_PUBLIC_WEBSITE_URL;
+
+  useEffect(() => {
+    if (isManagedEventType) {
+      form.setValue("metadata.managedEventConfig.unlockedFields", unlockedManagedEventTypeProps);
+    } else {
+      form.setValue("metadata", null);
+    }
+  }, [schedulingTypeWatch]);
+
+  const { register, setValue, formState } = form;
+
+  const createMutation = trpc.viewer.eventTypes.create.useMutation({
+    onSuccess: async ({ eventType }) => {
+      await utils.viewer.eventTypes.getByViewer.invalidate();
+      handleSuccessMutation(eventType);
+      form.reset();
+    },
+    onError: (err) => {
+      if (err instanceof HttpError) {
+        const message = `${err.statusCode}: ${err.message}`;
+        showToast(message, "error");
+      }
+
+      if (err.data?.code === "BAD_REQUEST") {
+        const message = `${err.data.code}: ${t("error_event_type_url_duplicate")}`;
+        showToast(message, "error");
+      }
+
+      if (err.data?.code === "UNAUTHORIZED") {
+        const message = `${err.data.code}: ${t("error_event_type_unauthorized_create")}`;
+        showToast(message, "error");
+      }
```

Comment:
Is it possible to refactor all this out in a custom hook to be shared across any event-type creation form.
Right now it is a duplicate of CreateEventTypeDialog code. It would make sense for the code to be more DRY.

Any other way that you can think of to make it DRY.

Reply:
Yeah, this makes sense. Thanks for the suggestion. I will fix that.